src/main/scala/com/github/blemale/scaffeine/Scaffeine.scala

```diff
@@ -166,6 +166,35 @@ case class Scaffeine[K, V](underlying: Caffeine[K, V]) {
     Scaffeine(underlying.expireAfterAccess(duration.toNanos, TimeUnit.NANOSECONDS))
 
   /**
+   * Specifies that each entry should be automatically removed from the cache once a duration has
+   * elapsed after the entry's creation, the most recent replacement of its value, or its last
+   * read.
+   *
+   * @param create the length of time an entry should be automatically removed from the cache after the entry's creation.
+   * @param update the length of time an entry should be automatically removed from the cache after the replacement of it's value.
+   * @param read   the length of time an entry should be automatically removed from the cache after the entry's last read.
+   * @tparam K1 the key type of the expiry.
+   * @tparam V1 the value type of the expiry.
+   * @return this builder instance
+   * @throws IllegalStateException if expiration was already set or used with expiresAfterAccess or expiresAfterWrite.
+   */
+  def expireAfter[K1 <: K, V1 <: V](
+    create: (K1, V1, Long) => Duration,
```

Comment:
In hindsight, the `currentTime` param may have been a mistake. Since the unit is nanoseconds and the Ticker uses System.nanoTime(), the value has no relation to the wall clock time. If the main usage would be to calculate the difference from a time stamp (e.g. db record's creation_date), then the parameter should not be used. Otherwise I think users would specify a fixed duration, so I'm unsure when the parameter would be useful. It was free to provide and I was more focused on the implementation details that I overlooked it, but now I'm afraid it may cause unnecessary confusion. If you match the API for consistency, you should stress this point as I'll do in my documentation updates.